client/src/pages/Catalog.tsx

```diff
@@ -27,23 +29,6 @@ export function Catalog() {
     getStyle();
   }, [searchParams]);
 
-  // useEffect(() => {
```

Comment:
Above, where it says `const style = searchParams.get('style') ?? '';`, this is good, but it ends up sending `style=` as a query parameter. Your server handles that just fine. However, it would be cleaner to remove the query parameter entirely if you aren't setting style. You can do that by computing the entire query string instead of just the style.

Something like:
```
const style = searchParams.get('style');
const query = style ? `?style=${style}` : '';
const response = await fetch(`/api/catalog${query}`);
```